api/server/app.cc

```diff
@@ -390,37 +390,55 @@ struct pg_error *pg_error;
 
 }  // namespace app
 
+
+
+const char *src_cgroup() {
+    if (!access("/sys/fs/cgroup/cpu/cpu.shares", R_OK)) {
+        return "/sys/fs/cgroup/cpu";
+    } else if (!access("/sys/fs/cgroup/cpu.shares", R_OK)) {
+        return "/sys/fs/cgroup";
+    }
+    return NULL;
+}
+
 int init_cgroup(int multiplier) {
-  system("mkdir /sys/fs/cgroup/cpu/butterfly");
-  system(std::string("echo $(( `cat /sys/fs/cgroup/cpu/cpu.shares` * "
-                     + std::to_string(multiplier)
-                     + " )) > /sys/fs/cgroup/cpu/butterfly/cpu.shares")
-        .c_str());
-  return 0;
+    system(std::string("mkdir " + std::string(src_cgroup()) +
+                       "/butterfly").c_str());
+    system(std::string("echo $(( `cat " +
+                       std::string(src_cgroup()) + "/cpu.shares` * "
```

Comment:
same here : check `src_cgroup()` and you may keep it's result at beginning

Also, you are not using bash, system calls `/bin/sh`